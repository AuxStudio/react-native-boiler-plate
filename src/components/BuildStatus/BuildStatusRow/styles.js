import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  text: {
    ...styleConstants.fonts.types.small,
  },
  boldText: {
    ...styleConstants.fonts.families.medium,
  },
});

export default styles;
