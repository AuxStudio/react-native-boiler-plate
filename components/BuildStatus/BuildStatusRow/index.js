import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

import styles from './styles';

const propTypes = {
  labelText: PropTypes.string,
  valueText: PropTypes.string,
};

const defaultProps = {};

const BuildStatusRow = ({ labelText, valueText }) => {
  return (
    <Text style={styles.text}>
      <Text style={styles.boldText}>{labelText} </Text>
      {valueText}
    </Text>
  );
};

BuildStatusRow.propTypes = propTypes;
BuildStatusRow.defaultProps = defaultProps;

export default BuildStatusRow;
