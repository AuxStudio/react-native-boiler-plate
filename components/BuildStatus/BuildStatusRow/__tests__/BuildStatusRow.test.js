import React from 'react';
import renderer from 'react-test-renderer';

import BuildStatusRow from '..';

describe('BuildStatusRow', () => {
  const labelText = 'Foo';
  const valueText = 'Bar';

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(
        <BuildStatusRow labelText={labelText} valueText={valueText} />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
