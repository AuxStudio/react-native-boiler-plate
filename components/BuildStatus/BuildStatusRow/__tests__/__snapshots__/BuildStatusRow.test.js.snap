// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildStatusRow renders renders 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    Foo
     
  </Text>
  Bar
</Text>
`;
